db/db_impl/db_impl_open.cc

```diff
@@ -533,8 +542,15 @@ Status DBImpl::Recover(
     // attention to it in case we are recovering a database
     // produced by an older version of rocksdb.
     std::vector<std::string> filenames;
+    std::vector<std::string>* files_in_wal_dir = &filenames;
     if (!immutable_db_options_.best_efforts_recovery) {
-      s = env_->GetChildren(immutable_db_options_.wal_dir, &filenames);
+      if (immutable_db_options_.wal_dir == dbname_) {
```

Comment:
At this point, we do not need to distinguish between `wal_dir == dbname_` and `wal_dir != dbname_`. We can just do
```
s = env_->GetChildren(wal_dir, &filenames);
```

Reply:
Note that `filenames` is in an inner scope invisible to the `if(read_only)` scope at the bottom. If we don't distinguish these two, then we'll have to move this variable to an outer scope in order to make it reusable.

Reply:
I think moving it to an outer scope sounds fine.